pkg/metrics/metricbuilder/metricbuilder_test.go

```diff
@@ -0,0 +1,78 @@
+package metricbuilder
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func getDescStruct(name string, desc string, constLabels map[string]string, variableLabels []string) string {
+	lpStrings := make([]string, 0, len(constLabels))
+	for key, value := range constLabels {
+		lpStrings = append(lpStrings, fmt.Sprintf("%s=%q", key, value))
+	}
+
+	return fmt.Sprintf("Desc{fqName: %q, help: %q, constLabels: {%s}, variableLabels: %v}",
+		name, desc, strings.Join(lpStrings, ","), variableLabels)
+}
+
+func TestMetricBuilderInvocation(t *testing.T) {
```

Comment:
rather than testing the individual metrics ("invocation", "duration," &c.) I think we should test the functionality of creating the prometheus types (gauge, histogram, counter). We should also include failed use cases, to show that we are capturing errors, e.g. create a histogram without buckets.